src/js/legend.js

```diff
@@ -0,0 +1,227 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.defaults("gpii.chartAuthoring.pieChart.legend", {
+        gradeNames: ["gpii.d3ViewComponent", "autoInit"],
+        strings: {
+            legendColHeading:"Legend",
+            labelColHeading:"Label",
+            valueColHeading:"Value"
+        },
+        model: {
+            // dataSet accepts:
+            // 1. an array of objects. Must contain "id", "value" and "label" variables.
+            // Example: [{id: string, value: number, label: string} ... ]
+            dataSet: []
+            // "dataSetWithColors" consolidates the dataset in original order with the array of colors, into a single array of objects
+            // dataSetWithColors: []
+        },
+        legendOptions: {
+            // An array of colors to fill slices generated for corresponding values of model.dataSet
+            // Or, a d3 color scale that's generated based off an array of colors
+            colors: null,
+            sort: true, // Whether or not to sort the data by values when creating the legend
+            showLegendHeadings: true // Whether or not to display column headings in the legend
+        },
+        modelRelay: [{
+            target: "dataSetWithColors",
+            singleTransform: {
+                type: "fluid.transforms.free",
```

Comment:
@waharnum, @cindyli and I had talked about this a bit before. I believe the approach we landed on was to create a generalized transformation that would handle adding a single property to a single object at a given path, or insert a value from an array of values into a corresponding object in an array of objects, at a given path that is the same for every object in the array. Essentially it'd be like using fluid.set but it could work on an array of objects and could take an array of values.

There is some older documentation on the GPII wiki for transformation ( https://wiki.gpii.net/w/Architecture_-_Available_transformation_functions ). This should cover most transformations and give you an idea of what they are and what they are for. To get an idea of what is currently in master, you'll have to read through the [modelTransformation code](https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/ModelTransformation.js), [modelTransformationTransforms code](https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/ModelTransformationTransforms.js), and [tests](https://github.com/fluid-project/infusion/blob/master/tests/framework-tests/core/js/ModelTransformationTests.js). We probably don't have to worry about writing inverse transforms for this right now.

You can also look at the other [transformations](https://github.com/fluid-project/chartAuthoring/blob/master/src/js/transformations.js) I wrote for chartAuthoring already.

Let me know if more clarification is needed.